src/persist-types/src/stats.rs

```diff
@@ -44,6 +44,141 @@ pub use structured::StructStats;
 
 include!(concat!(env!("OUT_DIR"), "/mz_persist_types.stats.rs"));
 
+/// Statistics for a single column of data.
+#[derive(Debug, Clone)]
+pub struct ColumnarStats {
+    /// Expected to be `None` if the associated column is non-nullable.
+    pub nulls: Option<ColumnNullStats>,
```

Comment:
How would you feel about inlining the null count here, and setting it to zero for a non-nullable type?

Seems like it could simplify some downstream code, and I'm not sure the distinction between `None` and `Some(0)` is buying us any safety...

Reply:
I nested the inner struct so we could differentiate between columns that could have nulls but does not contain any (`Some(0)`) vs non-nullable columns (`None`). I don't know if making that distinction is necessary though, so I'm more than happy to flatten it!

Reply:
> I don't know if making that distinction is necessary though [...]

Right - I suspected it's not, but you're the one who will know for sure!

Reply:
I'm not going to inline the null count for now, in a stacked PR being able to make the extra assertions is kind of nice. Can definitely change this later on though if we want!